fleetctl: check for empty input unit strings

dongsupark opened the pull request:
If no unit name is given for fleetctl commands like destroy, load,
start, stop, submit, and unload, then return with exit code 0,
printing out a soft warning message to stderr.

Fixes: https://github.com/coreos/fleet/issues/1443

/cc @kayrus @tixxdz @jonboulle

kayrus commented:
@dongsupark can you put these checks somewhere in one place. these copy-pastes look terrible.

dongsupark commented:
@kayrus I updated the branch dongsu/error-empty-units.
I wrote a simple wrapper in pkg/args.go, which can be called from each fleetctl command.

tixxdz commented:
@jonboulle lgtm

jonboulle commented:
uh, this helper is way overkill. please just revert to the inline check...

jonboulle commented:
```
	if pkg.IsArgsEmpty(args) {
		stderr("No units given")
		return 0
	}
```
is no simpler than
```
	if len(args) == 0 {
		stderr("No units given")
		return 0
	}
```

kayrus commented:
@jonboulle is it possible to check arguments inside `init` function?

jonboulle commented:
Why do you need to check inside init? In the context where you're checking, you've just been passed them.

dongsupark commented:
@jonboulle all right. I reverted it to the 1st version. So there's no helper in pkg/args.go.

jonboulle commented:
Thanks! Could you add a test for this? :-)

dongsupark commented:
@jonboulle Sure, I can.
Actually a PR by @tixxdz https://github.com/coreos/fleet/pull/1460 already contains a set of necessary tests for start and load. I would rather wait for the PR to be first merged, and add my tests on top of that.

jonboulle commented on fleetctl/destroy.go:
```diff
@@ -35,6 +35,13 @@ Destroyed units are impossible to start unless re-submitted.`,
 }
 
 func runDestroyUnits(args []string) (exit int) {
+	// Even if no unit is given by user, it should not exit with error,
```
I don't think the comment is necessary. Please just remove it here and everywhere else.

dongsupark commented:
Updated the tree as @jonboulle suggested.
I erase the comments as suggested.
I also added simple tests for destroy, load, start, stop, and unload.
Submit test is missing in this PR, because there's no fleetctl/submit_test.go.
So I created a new test submit_test.go that is covered in #1473.

tixxdz commented:
@dongsupark LGTM, but could you please create a new branch then force push on top of this so semaphore could run again ?

dongsupark commented:
Closed in favor of #1475
